docs/source/plugins.rst

```diff
@@ -0,0 +1,92 @@
+
+Writting plugins for **mokapot**
+================================
+
+This is a tutorial on how to write plugins for mokapot. 
+
+The plugin architecture allows you to extend mokapot without
+requiring any changes to the core of mokapot.
+
+Right not it allows you to modify the data before
+it gets used by the models or implement a new model.
+
+Plugins are python packages that export a `BasePlugin` object
+to an entry point called `mokapot.plugins`.
+So at thr very least you need to set that entry point up in your package configuration.
+
+If using setuptools with `setup.py` you can do that like this:
+
+    setup(
+        ...
+        entry_points={
+            'mokapot.plugins': [
+                'myplugin = my_package:MyPluginClass',
+            ],
+        },
+    )
+
+If using `pyproject.toml` you can do that like this:
+
+    [project.entry-points."mokapot.plugins"]
+    myplugin = "my_package:MyPluginClass"
+
+If using poetry you will need these lines in your `pyproject.toml`:
+
+    [tool.poetry.plugins."mokapot.plugins"]
+    "myplugin" = "my_package:MyPluginClass"
+
+
+An example plugin can be found in the tests for mokapot in
+`tests/system_tests/sample_plugin`
+
+The plugin needs to implement the `BasePlugin` class.
+You do not need to implement all the methods, but you can if you want to.
+
```

Comment:
**todo:** It would be really nice to expand on what the different methods do and the scope of what a plugin can do here.

Reply:
I feel like that gets covered below, do you feel like we should change the order?